### PR TITLE
Support Python 2.7, fixes #3

### LINKED ISSUE #3: Python 2 support
Trying to run the `animated` decorator example on Python 2, froze my machine. Any plans to support older versions of Python?

### PR DESCRIPTION
**Pull request by mkuznets**
* Explicit unicode handling (unicode_literals, decode string coming to Writting stream)
* Add arguments to super() (required in py2)
* Add with_metaclass() for universal metaclass support
* Redefine zip with its lazy version for py2
* .format_map(x) -> .format(**x)
* Rename λ argument in space_wave() (not supported in py2)
* Convert absolute imports to relative ones
* Call _over_wrapper directly from instance

### CODE REVIEW

File: tests/test_decorator.py [hunk 1]
```diff
@@ -40,12 +40,12 @@ def stop(self):
 
     def login(self):
         """Login in the wired with the default user"""
-        print('Welcome to the Wired, {user}!'.format_map(vars(self)))
```
Comment (ryukinix) [inline tests/test_decorator.py:43]: Please, you can rewrite this in the simple form `.format(user=self.user)`?

File: tests/test_decorator.py [hunk 2]
```diff
@@ -40,12 +40,12 @@ def stop(self):
...
+        print('Welcome to the Wired, {user}!'.format(**vars(self)))
 
     def logoff(self):
         """Exits of this world, open the next"""
         print('Close this world, open the next!')
-        print('Goodbye, {user}...'.format_map(vars(self)))
```
Comment (ryukinix) [inline tests/test_decorator.py:48]: The same on line 43

File: decorating/general.py [hunk 3]
```diff
@@ -15,6 +17,34 @@
...
+
+
+if PY2:
+    from itertools import izip
+    zip = izip
+else:
+    zip = zip
+
+
+def with_metaclass(meta, *bases):
```
Comment (ryukinix) [inline decorating/general.py:33]: This seems really trick, works fine on both versions?

File: decorating/decorator.py [hunk 4]
```diff
@@ -59,7 +60,7 @@ def stop(self):
             self.logoff()
 
         def login(self):
-            print('Welcome to the Wired, {user}!'.format_map(vars(self)))
+            print('Welcome to the Wired, {user}!'.format(**vars(self)))
```
Comment (ryukinix) [inline decorating/decorator.py]: Again, avoid `format(**)` and use simple form `format(user=self.user)`.

File: decorating/animation.py [hunk 5]
```diff
@@ -153,7 +151,7 @@ def _spinner(control):
                     padding=colorize_no_reset(padding, 'blue'),
                     start=colorize_no_reset(start, 'cyan'),
                     end=color.colorize(end, 'cyan'))
-        message = '\r' + template.format_map(info)
+        message = '\r' + template.format(**info)
```
Comment (ryukinix) [inline decorating/animation.py]: Well... just forgot the `info` variable and put inside of format.

**Review by ryukinix — CHANGES_REQUESTED** [review]
Thanks for PR!
I just suggest avoid this: * .format_map(x) -> .format(**x)

I wrote some comments as alternative for this.

**Review by ryukinix — APPROVED** [review]
I'm ok with it. Looks good. Thanks again.

**Comment by mkuznets** [comment]
You're welcome :-)

**Pull request merged**